### PR TITLE
Style fixes & consistency, updates for 2.0

### PR DESCRIPTION
**Pull request by valeriangalliat**
* [x] Typography consistency.
* [x] Configuration examples in YAML instead of JSON, except final JSON example.
* [x] Data examples in JS, respecting our [guidelines](https://github.com/SassDoc/sassdoc/blob/develop/GUIDELINES.md).
* [x] Update parts of the API examples to 2.0.
* [x] Update documentation on how `package` and `theme` are resolved.
* [x] Mention `--parse` flag on data interface page (can help people to see actual generated data and understand it).
* [x] Remove mention to `--sass-convert`, update to new behavior.
* [x] Ensure double spacing in HTML classes.
* [x] Update part of theme page to Themeleon 3.0 behavior.
* [x] `/configuration/#autofill`: list annotations that support autofill and how to configure it (do I include the `@` for example?).
* Homepage:
  * [x] "so you end up with a fully styled HTML document, like this"; update screenshot (still showing 1.0).
  * [x] "All these projects are built with SassDoc"; "are build with sassdoc" sounds odd to me; the projects are not built with SassDoc, their documentation is built with SassDoc. Not sure it's necessary to distinguish, it's somehow obvious.
* [x] `/gallery/` killed my PC, need to optimize (like use compressed images, and not in full size).
* [x] `/gallery/`: ensure Bourbon Neat docs looks good by injecting custom CSS in webshot, to put a hard main content width (PhantomJS is really buggy with SVGs).
* [x] Using your own theme is not linked too. It could be linked from configuration page "these options are for default theme but you can use your own theme", and in Getting started with "`--theme` defaults to `default` but you can use your own theme".
* [x] Check if we need to update `generator-sassdoc-theme` and `sassdoc-theme-blank` for 2.0.
* Using your own theme:
  * [x] "Note: you are not required to use Swig. Jade, Mustache, Nunjucks and Handlebars are also available"; now everything is available through consolidate.js, need to change this sentence.
  * [x] "SassDoc will try to require the following until one matches"; this is a duplicate with `/configuration/#theme`, maybe just link to it?
  * [x] Update exemples with consolidate.js (meaning update `package.json` examples too, because the template engine has to be explicitely required).
  * [x] Update Swig extras examples, because of consolidate.js the global Swig instance needs to be modified.
  * [x] Link to the theme generator when speaking about it.
* [x] `/node/`: mention stream usage, and say that's why it works out of the box in Gulp (and link to Gulp page).
* <del>`/about/`: profile image for contributor could be an octopus, to stay in SassDoc style instead of the current neutral image.</del>
* [x] Theme gallery is not linked anywhere. It could be linked near `--theme` option, and maybe in Using your own theme with "you can inspirate from other themes".
* [x] `/getting-started/`: mention stdin/stdout CLI usage (without source argument).

### CODE REVIEW

**Comment by HugoGiraudel** [comment]
Yes, that is awesome

**Comment by valeriangalliat** [comment]
BTW there's some question in the list, @SassDoc/owners if any of you already know the answer it can avoid me some investigation on SassDoc.

> /configuration/#autofill: list annotations that support autofill and how to configure it (do I include the @ for example?).

>  On homepage, "All these projects are built with SassDoc"; maybe include some names/logos below, not just a link? 

Opinions on this?

> "are build with sassdoc" sounds odd to me; the projects are not built with SassDoc, their documentation is built with SassDoc. Not sure it's necessary to distinguish, it's somehow obvious.

And this?

>  /about/: profile image for contributor could be an octopus, to stay in SassDoc style instead of the current neutral image.

Thoughts about this?

>  /contributing/: the link to FAQ fails for some reason.

@HugoGiraudel, do we just need to add FAQ to the references include? I'm not confident with this.

**Comment by HugoGiraudel** [comment]
> Someone knows how autofill works?

@FWeinb.

> On homepage, "All these projects are built with SassDoc"; maybe include some names/logos below, not just a link?

Most projects do not have any logo.

> "are build with sassdoc" sounds odd to me; the projects are not built with SassDoc, their documentation is built with SassDoc. Not sure it's necessary to distinguish, it's somehow obvious.

First, it should be *are built*. Feel free to replace with *documented with SassDoc*.

> /about/: profile image for contributor could be an octopus, to stay in SassDoc style instead of the current neutral image.

Why not. Maybe our illustration or something. Minor detail though.

> /contributing/: the link to FAQ fails for some reason.

We have it already: https://github.com/SassDoc/sassdoc.github.io/blob/develop/_data/routes.yml#L19.

**Comment by valeriangalliat** [comment]
My bad for FAQ link, it was a problem with non-closed HTML tag, and the routes were not included. Already fixed.

> First, it should be are built. Feel free to replace with documented with SassDoc.

Yeah "are built" is gramatically right but it's the distinction with "build software" and "document software" that was bugging me. I think I'll go with "documented with", thanks.

**Comment by FWeinb** [comment]
`autofill` takes an array of (none alias) names of annotations to autofill.

**Comment by valeriangalliat** [comment]
@FWeinb Currently, only `@throws` and `@requires` can be autofilled right? I want to include a list of possible annotations to include in `autofill`.

So the following is a valid example?

```js
{
  autofill: ['throws', 'requires'],
}
```

**Comment by FWeinb** [comment]
@valeriangalliat `content`

File: _data/gallery.yml [hunk 1]
```diff
@@ -22,9 +22,9 @@
   url: http://madebymany.github.io/css-patterns/
   image: css-patterns.png
 
-- name: scss-components
-  url: http://scss-components.com/sassdoc/
-  image: scss-components.png
+# - name: scss-components
```
Comment (HugoGiraudel) [inline _data/gallery.yml]: Why is this project commented?

File: configuration/index.md [hunk 2]
```diff
@@ -61,25 +58,27 @@ If this doesn't work for you, then you can set your own package. The package obj
 * `homepage`: URL to your project's homepage
 * `description`: description of your project
 
-<p class="note note--info">
+<p class="note  note--info">
```
Comment (HugoGiraudel) [inline configuration/index.md]: ERMHAGERD!! PATCH OF THE DAY!

File: configuration/index.md [hunk 3]
```diff
@@ -43,14 +43,11 @@ SassDoc will wipe this directory at every run, so don't put anything worthwhile
...
+The `package` option is either a path (string) to a JSON file or directly an object.
 It contains information about the documented project (for instance a `package.json`, hence the name of the key).
 
-Please note that we try to find a `package.json` in 3 different locations before giving up:
-
-* a relative path from your current folder
-* a relative path from the theme folder
-* an absolute path.
+If you provide a path, it will be resolved from the configuration file (or CWD if you have no configuration file).
+By default, it's `package.json` (with the same resolving rules).
```
Comment (HugoGiraudel) [inline configuration/index.md]: Don't line break here.

File: data-interface/index.md [hunk 4]
```diff
@@ -9,6 +9,8 @@ SassDoc is doing quite a lot of things under the hood. In most cases, you won't
 
 <p class="note  note--info"><strong>Clarification:</strong> in the following examples, we are using C-style comments but it works exactly the same for inline comments as well. Please refer to the <a href="/annotations/">Annotations</a> page for the syntax.</p>
 
+<p class="note  note--info"><strong>Note:</strong>You can run SassDoc in parse mode with the <code>--parse</code> CLI flag to output the data as JSON instead of generating a documentation directory.</p>
```
Comment (HugoGiraudel) [inline data-interface/index.md]: I was not aware of that. :x

File: _data/gallery.yml (hunk 1 above)
Comment (valeriangalliat) [inline _data/gallery.yml]: The website don't exists anymore and it fails to generate the screenshot.

File: data-interface/index.md [hunk 5]
```diff
@@ -21,32 +23,30 @@ Note that as a matter of simplicity, all annotations are mapped to an array, eve
...
+    code: '',
+    line: {
+      start: 0,
+      end: 0,
+    },
   },
-  'access': ['public'],
-  'group': [['undefined']]
+  access: ['public'],
+  group: [['undefined']],
```
Comment (HugoGiraudel) [inline data-interface/index.md]: Are we still using two nested arrays for `group`? Ping @FWeinb.

File: grunt/index.md [hunk 6]
```diff
@@ -62,27 +62,27 @@ in the repository for a full example.
...
-      'src': 'path/to/sass',
-      'dest': 'path/to/docs',
-      'options': {
-        'config': 'path/to/config.json'
-      }
-    }
-  }
+  sassdoc: {
+    default: {
+      'src: path/to/sass',
```
Comment (HugoGiraudel) [inline grunt/index.md]: Issue with quotes here.

File: configuration/index.md (hunk 3 above)
Comment (valeriangalliat) [inline configuration/index.md]: I only join lines 49/50? Or also 41/42 and 46/47?

File: gulp/index.md [hunk 7]
```diff
@@ -9,7 +9,7 @@ This is a [Gulp friendly](https://github.com/gulpjs/gulp/blob/master/docs/writin
 Its goal is to make SassDoc integration in your Gulp workflow easier.
 plugin, not using the benefits of streams.
 
-If you haven't used [gulp](http://gulpjs.com) before, be sure to check out the
+If you haven't used [Gulp](http://gulpjs.com) before, be sure to check out the
```
Comment (HugoGiraudel) [inline gulp/index.md]: This whole page needs to be rewritten to deprecate the plugin.

File: data-interface/index.md (hunk 4 above)
Comment (valeriangalliat) [inline data-interface/index.md]: That was *also* the purpose of documentation. :smile_cat:

File: data-interface/index.md (hunk 5 above)
Comment (valeriangalliat) [inline data-interface/index.md]: According to `--parse`, no.

```json
    "group": [
      "cross-browser-support"
    ],
```

I'll update the doc.

File: gulp/index.md (hunk 7 above)
Comment (valeriangalliat) [inline gulp/index.md]: Yep, there's a bunch of @pascalduez issues for this.

File: node/index.md [hunk 8]
```diff
@@ -21,30 +21,62 @@ It's up to you to do whatever you want with this!
...
+
+sassdoc(source)
+  .then(function () {
+    console.log('Your documentation has been generated!');
+  }, function (err) {
+    console.error(err);
+  });
+{% endhighlight %}
+
+If you want to [configure][configuration] the destination directory, you can pass a custom `dest` in the configuration object.
```
Comment (HugoGiraudel) [inline node/index.md]: There is no mention of the configuration object in the previous example.

File: node/index.md [hunk 9]
```diff
@@ -21,30 +21,62 @@ It's up to you to do whatever you want with this!
...
+
+Both `sassdoc` and `sassdoc.parse` functions have the same synopsis:
+
+* `@param {String} source` path to source or [glob string/array](#about-glob);
+* `@param {Object} [configuration]` optional [configuration] object;
+* `@return {Promise}`.
+
+## Streaming
+
+I lied to you in the [Synopsis](#synopsis) part. There is one more way you can use these functions; if you ommit the `source` argument, you'll get a [transform stream][repo_through2], able to handle [vinyl files][repo_vinyl]. You can easily use it with [vinyl-fs][repo_vinyl_fs].
```
Comment (HugoGiraudel) [inline node/index.md]: No *I*.

Comment (valeriangalliat) [inline node/index.md]: WAT?

File: _data/gallery.yml (hunk 1 above)
Comment (HugoGiraudel) [inline _data/gallery.yml]: Alright. Then remove it completely.

File: configuration/index.md (hunk 3 above)
Comment (HugoGiraudel) [inline configuration/index.md]: 49 and 50.

File: node/index.md (hunk 9 above)
Comment (HugoGiraudel) [inline node/index.md]: Nowhere else in the docs do we speak using *I*. Don't use *I*. Use *we* at worst, no pronoun at best.

Comment (valeriangalliat) [inline node/index.md]: Oh sorry I saw a slash while it's an italic `I`. :/

**Comment by HugoGiraudel** [comment]
LGTM.

**Pull request merged**